FFM-7755: reorders sections of tutorial (#1609)
harness/developer-hub: tutorials/feature-flags/typescript-react.md

## Background on Feature Flags

> As software engineers, we innovate by diving into creating the unknown, sometimes for the first time. Core to innovative work is experimenting and prototyping, because incremental success builds greater success. As changes get closer to production, the experimentation that was appropriate in lower environments tapers off as the potential blast radius grows. That is, if we get the experiment wrong, there’s a greater impact on a larger number of users.

Reviewer: Do we need this paragraph? Sounds too fancy like a marketing material for a technical tutorial.

Author: At the moment, this tutorial is definitely used for marketing purposes, and the request was by product people for that purpose, so I'm not going to change too much about the tutorial for now.

> [Feature Flags](https://harness.io/blog/what-are-feature-flags) lets you experiment by allowing blocks of code to be executed conditionally. Conditional statements are nothing new, but the management of inventorying and toggling these conditional statements are newer, and Feature Flag platforms are here. Feature Flags enable progressive delivery, or the incremental release of features. 

Author: agree, but not wordsmithing too much for this

In this tutorial we will walk you through your first Feature Flag creation and execution.

> ## What you need For executing your first Feature Flag

Reviewer: ```suggestion
## Requirements
```

Author: leaving marketing-speak

> This example will require a GitHub Account, an editor such as an IDE, a [Harness Feature Flags Account](https://app.harness.io/auth/#/signup/?module=cf&?utm_source=website&utm_medium=harness-developer-hub&utm_campaign=ff-plg&utm_content=get-started), and the ability to run an [NPM Install](https://docs.npmjs.com/cli/v7/commands/npm-install) in your local or remote environment.

Reviewer: Make sure you have the following:
* A GitHub account.
* An editor such as IDE.
* A [Harness Feature Flags Account](https://app.harness.io/auth/#/signup/?module=cf&?utm_source=website&utm_medium=harness-developer-hub&utm_campaign=ff-plg&utm_content=get-started).
* The ability to run [NPM Install](https://docs.npmjs.com/cli/v7/commands/npm-install) in your local or remote environment.

![Feature Flag Overview](static/ff-tutorial-typescript/ff_overview.png)

> Feature Flags have a close relationship with the code base since they surround/decorate conditional statements. You’ll need to include a Feature Flag SDK as a dependency, then wire the Feature Flag client. To support this, the underlying code base and necessary dependencies to build/run the code base is needed. If this is your first time leveraging a Feature Flag, [Google’s React Store](https://github.com/google-pay/react-store) is a great sample app to modify and use with Harness Feature Flags. 

Reviewer: ```suggestion
Feature Flags have a close relationship with the code base as they use conditional statements. You must include a Feature Flag SDK as a dependency, then wire the Feature Flag client. To support this, the underlying code base and necessary dependencies to build or run the code base are needed. If this is your first time leveraging a Feature Flag, [Google’s React Store](https://github.com/google-pay/react-store) is a great sample app to modify and use with Harness Feature Flags. 
```

Author: good suggestions, but again I'm leaving the tone as is right now.
